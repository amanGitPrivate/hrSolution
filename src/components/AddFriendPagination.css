:local(.friendListPagination)
{
    width: 50%;
    display: flex;
    justify-content: space-around;
    align-items: center;
}
