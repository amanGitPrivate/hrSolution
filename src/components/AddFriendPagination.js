import React from 'react';
import PropTypes from 'prop-types';
import styles from './AddFriendPagination.css';
import classnames from 'classnames';

const AddFriendPagination = props => {
  const {
    pageNumber,
    friends,
    itemsPerPage,
    goToPrevPage,
    isFirstPage,
    goToNextPage,
    isLastPage
  } = props;
  return (
    <div className={classnames(styles.friendListPagination)}>
      <button
        onClick={() => goToPrevPage()}
        disabled={isFirstPage()}
        className="btn btn-default"
      >
        <i className="fa fa-caret-left" />
      </button>
      <label className={styles.friendListPaginationPageNumber}>
        {pageNumber} of {Math.ceil(friends.length / itemsPerPage)}
      </label>
      <button
        onClick={() => goToNextPage()}
        disabled={isLastPage()}
        className="btn btn-default"
      >
        <i className="fa fa-caret-right" />
      </button>
    </div>
  );
};

AddFriendPagination.propTypes = {
  pageNumber: PropTypes.number,
  friends: PropTypes.array,
  itemsPerPage: PropTypes.number,
  goToPrevPage: PropTypes.func.isRequired,
  isFirstPage: PropTypes.func.isRequired,
  goToNextPage: PropTypes.func.isRequired,
  isLastPage: PropTypes.func.isRequired
};

export default AddFriendPagination;
