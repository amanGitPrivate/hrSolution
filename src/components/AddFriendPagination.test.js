import React from 'react';
import AddFriendPagination from './AddFriendPagination';
import renderer from 'react-test-renderer';

const friends = [
  {
    name: 'Theodore Roosevelt',
    starred: true
  },
  {
    name: 'Abraham Lincoln',
    starred: false
  },
  {
    name: 'George Washington',
    starred: false
  }
];

test('AddFriendPagination renders correctly', () => {
  const component = renderer.create(
    <AddFriendPagination
      pageNumber={1}
      friends={friends}
      itemsPerPage={2}
      goToPrevPage={jest.fn()}
      isFirstPage={jest.fn()}
      goToNextPage={jest.fn()}
      isLastPage={jest.fn()}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
