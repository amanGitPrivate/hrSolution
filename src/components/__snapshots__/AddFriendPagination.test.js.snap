exports[`test AddFriendPagination renders correctly 1`] = `
<div
  className="">
  <button
    className="btn btn-default"
    disabled={undefined}
    onClick={[Function]}>
    <i
      className="fa fa-caret-left" />
  </button>
  <label
    className={undefined}>
    1
     of 
    2
  </label>
  <button
    className="btn btn-default"
    disabled={undefined}
    onClick={[Function]}>
    <i
      className="fa fa-caret-right" />
  </button>
</div>
`;
